### ISSUE TITLE
CE after fix_includes.py

### ISSUE THREAD

**Issue by arseny30** [include-what-you-use/include-what-you-use#412]
a.cpp:
```
template <class T>
struct Owner {
  using AliasT = T;
  ~Owner();
};

template <class T>
Owner<T>::~Owner() {
}

struct Query {
  class Callback {};
  Owner<Callback> callback_;
};

int main(void) {
  return 0;
}
```
Run with:
`iwyu --std=c++11 a.cpp `

Output:
```
The full include-list for a.cpp:
class Query::Callback;
---
```

This all leads to a CE after suggested fixes are applied.
Also this code looks like a minimal test for me. All further test simplifications I've tried are fixing the problem.

I used `include-what-you-use 0.7 (git:136736f) based on clang version 3.9.0 (tags/RELEASE_390/final)`

**Comment by kimgr** [comment]
Thanks for the report!

Interesting case, since there are no `#include` directives anywhere :-)

I think the problem might be in `iwyu_output.cc`, the function called `DeclIsVisibleToUseInSameFile`. It looks like it doesn't take template uses into account.

**Comment by kimgr** [comment]
Fixed in PR #433.
